open_samus_returns_rando/specific_patches/map_icons.py

```diff
@@ -0,0 +1,332 @@
+from mercury_engine_data_structures.formats import Bmsmsd
+
+from open_samus_returns_rando.patcher_editor import PatcherEditor
+
+
+def _surface_icons(editor: PatcherEditor):
+    surface = editor.get_file("gui/minimaps/c10_samus/s000_surface.bmsmsd", Bmsmsd)
+    mapicon = surface.raw["tiles"]
+
+    # Doors
+    # Door004, Door011 (Chozo Seal Charge Doors)
+    mapicon[99]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[100]["icons"][0]["icon"] = "doorpowerright"
+
+    mapicon[57]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[58]["icons"][0]["icon"] = "doorpowerright"
+
+    for tile in range(479):
+        # Hazardous Tiles
+        if mapicon[tile]["tile_type"] == "ACID":
+            mapicon[tile]["tile_type"] = "NORMAL"
+        # Items
+        if mapicon[tile]["icons"] != ([]) and (
+            mapicon[tile]["icons"][0]["icon"].endswith("tank")
+            or mapicon[tile]["icons"][0]["icon"].startswith("itemsphere")
+            or mapicon[tile]["icons"][0]["icon"].startswith("spenergy")
+        ):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+
+
+def _area1_icons(editor: PatcherEditor):
+    area1 = editor.get_file("gui/minimaps/c10_samus/s010_area1.bmsmsd", Bmsmsd)
+    mapicon = area1.raw["tiles"]
+
+    # Doors
+    # Door002 (Chozo Seal Charge Door)
+    mapicon[77]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[78]["icons"][0]["icon"] = "doorpowerright"
+
+    # Door004 (Bombs)
+    mapicon[194]["icons"][0]["icon"] = "doorpowerleft"
+
+    # Door012 (Right Exterior Door -> Interior)
+    mapicon[266]["icons"][0]["clear_condition"] = ""
+    mapicon[267]["icons"][0]["icon"] = "doorpowerright"
+
+    # Door016 (Exterior Alpha)
+    mapicon[302]["icons"][0]["icon"] = "doorpowerright"
+
+    for tile in range(384):
+        # Hazardous Tiles
+        if mapicon[tile]["tile_type"] == "ACID":
+            mapicon[tile]["tile_type"] = "NORMAL"
+        # Items
+        if mapicon[tile]["icons"] != ([]) and mapicon[tile]["icons"][0][
+            "icon"
+        ].startswith("item"):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+    mapicon[158]["icons"][0]["icon"] = "itemenabledheat"
```

Comment:
I'm wondering: If the `tile_type` is already heat, what is the difference between `itemenabled` and `itemenabledheat`?

Reply:
Yeah, but what is the difference? Do they have a different color or why are these two different icons? Is the open circle not transparent that it would look blue within the red heat tile?

Reply:
The former is blue and the latter is red. I specified heat afterwards because it first sets to blue which is wrong when on the heat tile.